rust-runtime/inlineable/src/endpoint_lib/parse_url.rs

```diff
@@ -7,13 +7,23 @@ use crate::endpoint_lib::diagnostic::DiagnosticCollector;
 use http::Uri;
 use url::{Host, Url as ParsedUrl};
 
-#[derive(PartialEq, Debug)]
+#[derive(PartialEq, Debug, Clone)]
 pub(crate) struct Url<'a> {
     uri: Uri,
     url: ParsedUrl,
     raw: &'a str,
 }
 
+impl Default for Url<'_> {
```

Comment:
Not sure default makes sense here either. Presumably we are abusing `unwrap_or_default()` or something for this? Do we not have other options?